olxcleaner/loader/xml.py

```diff
@@ -239,15 +237,36 @@ def read_course(edxobj, node, directory, filename, errorstore, htmlfiles, pointe
 
         # Recurse into each child
         for child in node:
-            if child.tag is etree.Comment:
-                # Ignore comments
-                pass
-            elif child.tag in edxobj.allowed_children:
-                # Recurse on that node
-                newobj = EdxObject.get_object(child.tag)
-                edxobj.add_child(newobj)
-                read_course(newobj, child, directory, filename, errorstore, htmlfiles)
+            child_tag = child.tag
+            tag_is_allowed = check_tag_is_allowed(child_tag, edxobj.allowed_children, allowed_xblocks)
+            tag_is_comment = child_tag is etree.Comment  # Ignore comments
+            if tag_is_comment:
+                continue
+            elif tag_is_allowed:
+                try:
+                    newobj = EdxObject.get_object(child_tag)
+                    edxobj.add_child(newobj)
+                    read_course(newobj, child, directory, filename, errorstore, htmlfiles,
+                                allowed_xblocks=allowed_xblocks)
+                except ClassDoesNotExist:
+                    # Do not do anything if an xblock is allowed but currently
+                    # not supported by the library
+                    pass
             else:
-                errorstore.add_error(UnexpectedTag(filename,
-                                                   tag=child.tag,
-                                                   edxobj=edxobj))
+                errorstore.add_error(
+                    UnexpectedTag(filename, tag=child_tag, edxobj=edxobj)
+                )
+
+
+def check_tag_is_allowed(tag, allowed_children, allowed_xblocks):
+    """
+    Checks if a tag is valid or not.
+
+    :param tag: Tag name of the lxml element
+    :param allowed_children: List of xblocks which this tag can have
+    :param allowed_xblocks: List of all allowed xblocks which may or may not be supported currently.
+    """
+    tag_is_allowed = allowed_xblocks is not None and tag in allowed_xblocks
+    if allowed_xblocks is None:
```

Comment:
If we check this condition first there will be no need to check for empty in the above line.
```
if allowed_xblocks is None:
    return tag in allowed_children
else:
    return tag in allowed_xblocks
```

Reply:
Done.